src/data/sym2.lean

```diff
@@ -94,6 +94,19 @@ begin
   { cases h; rw [h.1, h.2], rw eq_swap }
 end
 
+/-- The universal property of `sym2`; symmetric functions of two arguments are equivalent to
+functions from `sym2`. Note that when `β` is `Prop`, it can sometimes be more convenient to use
+`sym2.from_rel` instead. -/
+def lift {β : Type*} : {f : α → α → β // ∀ a₁ a₂, f a₁ a₂ = f a₂ a₁} ≃ (sym2 α → β) :=
+{ to_fun := λ f, quotient.lift (uncurry ↑f) $ by { rintro _ _ ⟨⟩, exacts [rfl, f.prop _ _] },
+  inv_fun := λ F, ⟨λ a₁ a₂, F ⟦(a₁, a₂)⟧, λ a₁ a₂, congr_arg F eq_swap⟩,
```

Comment:
If we want to maintain the point-free style of `to_fun`, then `inv_fun `could be
```lean
  inv_fun := λ F, ⟨curry (F ∘ quotient.mk), λ a₁ a₂, congr_arg F eq_swap⟩,
```

Reply:
Sure, why not. I added a lemma that unfolds it to my version, but it's nice to see the composition in `lift.inv_fun`, since there almost always is one.